lib/ext/htmlentities/android_xml.rb

```diff
@@ -0,0 +1,23 @@
+require 'htmlentities'
+
+class HTMLEntities
+  MAPPINGS['android_xml'] = MAPPINGS['xhtml1'].dup.tap do |mappings|
+    mappings.delete('apos')
+  end
+
+  FLAVORS << 'android_xml'
+
+  class AndroidXmlEncoder < Encoder
+    def initialize(instructions = [])
+      super('android_xml', instructions)
+    end
+
+    private
+
+    # had to adjust this so that single and double quotes don't get turned into
+    # entities (they're escaped by hand with backslashes)
+    def basic_entity_regexp
+      @basic_entity_regexp ||= /[<>&]/
```

Comment:
This is just a small subset of characters that, for some reason, the [htmlentities](https://github.com/threedaymonk/htmlentities) gem encodes manually (instead of using the [`MAPPINGS` constant](https://github.com/threedaymonk/htmlentities/blob/master/lib/htmlentities/flavors.rb#L3)). I have the `AndroidXmlEncoder` inheriting from htmlentities' `Encoder` class to get all the mapping stuff for free and to pull out single/double quotes from the `@basic_entity_regexp`. It shouldn't mess with any characters that are already escaped because the code above removes 'apos' from the mapping and there's no special encoding for double quotes. Also, the htmlentities gem is smart enough to know to not escape `&` if it's part of `&amp;`.